### PR TITLE
Add additional theme

### PR DESCRIPTION
**Pull request by MarkBlyth**
Bristol university theme for typst slides, inspired by a previous version of [Barton's beamer theme](https://github.com/dawbarton/UoB-beamer-theme). While the branding is for Bristol University, the logos can easily be swapped out for another institution.

### CODE REVIEW

File: themes/uob.typ [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+// https://github.com/dawbarton/UoB-beamer-theme
+// =============================================
+
+#let section = state("section", none)
+#let subslide = counter("subslide")
+#let logical-slide = counter("logical-slide")
+#let repetitions = counter("repetitions")
+#let global-theme = state("global-theme", none)
+
+#let new-section(name) = section.update(name)
```
Comment (andreasKroepelin) [inline themes/uob.typ:14]: This need (and should) not be repeated here, user will write `#import "slides.typ": *` anyways.

File: themes/uob.typ [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+#let uob-theme(color: rgb(171, 31, 45)) = data => {
+    let title-slide = {
+
+     	place(image("UoB-watermark.svg", width:100%))
```
Comment (andreasKroepelin) [inline themes/uob.typ:19]: The image path should be a parameter.

File: themes/uob.typ [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+
+        v(5%)
+	grid(columns: (5%, 1fr, 1fr, 5%),
+	    [],
+	    align(bottom + left)[#image("UoB-logo.svg", width:40%)],
+	    align(bottom + right)[#image("UoB-url.svg", width:40%)],
```
Comment (andreasKroepelin) [inline themes/uob.typ:25]: The image paths should be parameters.

File: themes/uob.typ [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+                ..data.authors
+            )
+            #v(1em)
+            #data.date
+        ]
+    }
+
+    let default(slide-info, bodies) = {
+        if bodies.len() != 1 {
+            panic("default variant of default theme only supports one body per slide")
```
Comment (andreasKroepelin) [inline themes/uob.typ:58]: Adapt name of theme in error message

File: themes/uob.typ [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+        // footer
+        decoration("footer")[
+            #text(color, .6em)[#data.short-authors] #h(1fr)
+            #text(color, .6em)[#logical-slide.display()]
+        ]
+    }
+
+    let wake-up(slide-info, bodies) = {
+        if bodies.len() != 1 {
+            panic("wake up variant of default theme only supports one body per slide")
```
Comment (andreasKroepelin) [inline themes/uob.typ:116]: Adapt name of theme in error message.

File: themes/UoB-logo.svg
Comment (andreasKroepelin) [inline themes/UoB-logo.svg:1]: If we include these in this repo they should be in an extra directory, like `themes/uob`. Same for the other SVGs.

**Review by andreasKroepelin — COMMENTED** [review]
Thanks again for this nice theme! As next steps, we need to add a section to the [theme gallery](https://andreaskroepelin.github.io/typst-slides/book/theme-gallery/index.html), including a minimal demo.

File: themes/uob.typ (hunk 1 above)
Comment (andreasKroepelin) [reply themes/uob.typ]: Oh sorry, I didn't mean importing it, I mean that the usage will be
```typ
#import "slides.typ": *
#import "themes/uob.typ": *

#show: slides.with(...)
...
```
So then all of those globals are already in scope.

**Comment by MarkBlyth** [comment]
Thanks for the review, suggested changes have been made, including an entry in the book,

File: themes/uob.typ (hunk 1 above)
Comment (MarkBlyth) [reply themes/uob.typ]: Turns out we don't get a circular dependency by importing slides.typ into uob.typ, so we can get the variables in scope within uob.typ with an import. Implemented that now.

**Comment by MarkBlyth** [comment]
The current version now gives easy creation of multi-column slides, however it requires the most recent version of Typst to run correctly.

**Review by andreasKroepelin — COMMENTED** [review]
I'm sorry, I just realised I never finished my last review. Sorry that you had to wait!

**Comment by andreasKroepelin** [comment]
I just had a thought: How about renaming the theme "bristol"? I think that's better to keep in mind than a (for a non-Bristol user) random acronym. After all, beamer themes are also named after cities.

**Comment by MarkBlyth** [comment]
Definitely makes sense to give the theme a clearer name. Updated that now.

File: themes/uob/uob.typ [hunk 6]
```diff
@@ -5,13 +5,7 @@
...
 // =============================================
 
-#let section = state("section", none)
-#let subslide = counter("subslide")
-#let logical-slide = counter("logical-slide")
-#let repetitions = counter("repetitions")
-#let global-theme = state("global-theme", none)
-
-#let new-section(name) = section.update(name)
+#import "../../slides.typ": *
```
[the remark this answers is not in the document]
Comment (MarkBlyth) [reply themes/uob/uob.typ]: The relevant variables will only be in scope within the theme file if they are either imported into the theme file, or defined there in the first place. Even when the user imports both files into their document, the necessary variables won't be in scope in the theme file, so we need to have either an #import or #let in the theme file, as done here.

**Comment by andreasKroepelin** [comment]
The code looks pretty good, imo. Can you please take a look at https://andreaskroepelin.github.io/typst-slides/book/themes.html#create-your-own-theme again? The variant handling changed a bit since #26 and would need to adapt your code. It's just a small fix regarding the title slide and the returned dictionary, though.

**Comment by MarkBlyth** [comment]
Updated the template now, for the new returns format.

File: book/src/theme-gallery/index.md [hunk 7]
```diff
@@ -114,3 +115,74 @@ It features a dominant partition of space into a bright and a dark side.
...
 
+
+## Bristol
+_[go to top](#theme-gallery)_
+
+This is a variation of the default theme, to feature academic branding. It is inspired by an old version of David Barton's
+[Beamer theme](https://github.com/dawbarton/UoB-beamer-theme).
+It features a University of Bristol branding by default, however the logos and colour choices can easily be swapped, to tailor the theme to any institution of your choice.
+```typ
+#import "themes/bipartite.typ": *
```
Comment (andreasKroepelin) [inline book/src/theme-gallery/index.md:126]: Change `bipartite` to `bristol`

File: book/src/theme-gallery/index.md [hunk 8]
```diff
@@ -114,3 +115,74 @@ It features a dominant partition of space into a bright and a dark side.
...
+#show: slides(
+  // ...
+  theme: bristol-theme(),
+)
+```
+
+### Options
+- `color`: the colour to use for decorative elements, default University of Bristol red
+- `watermark`: file path for a watermark image to span the title slide, default "bristol/watermark.svg"
+- `logo`: file path for a logo image to appear on every slide, default "bristol/logo.svg"
```
Comment (andreasKroepelin) [inline book/src/theme-gallery/index.md:136]: Maybe note that those default files are shipped with the theme so users don't worry about providing them themselves.

File: book/src/theme-gallery/index.md [hunk 9]
```diff
@@ -114,3 +115,74 @@ It features a dominant partition of space into a bright and a dark side.
...
+- `secondlogo`: file path for an additional logo image, to appear on the first slide only, default "bristol/secondlogo.svg"
+
+### Variants
+- `title slide`: shows a nicely formatted title slide
+- `wake up`: no decoration, colored background, enlarged text
+
+### Extra keyword arguments
+- `title`: optional, a title for that slide
+- `colwidths` : optional, array specifying width of each column; default `1fr` per column
+- `gutter` : optional, integer, relative length, fraction, or array, specifying the gap between columns if a multicolumn (multi-body) slide is chosen; default 0.2em
```
Comment (andreasKroepelin) [inline book/src/theme-gallery/index.md:146]: Consider writing 0.2em as `0.2em`, i.e. as code

File: themes/bristol.typ [hunk 10]
```diff
@@ -0,0 +1,148 @@
+//==============================================
+// University of Bristol theme for Typst slides.
+// Based on a previous version of David Barton's 
+// UoB LaTeX Beamer template, found at
+// https://github.com/dawbarton/UoB-beamer-theme
+// =============================================
+
+#import "../slides.typ": *
```
Comment (andreasKroepelin) [inline themes/bristol.typ:8]: This should not be necessary?

**Review by andreasKroepelin — COMMENTED** [review]
Alright, I think we're pretty much done here. I just found a couple of small things but then I think the theme should be ready to be merged!

File: themes/bristol.typ (hunk 10 above)
Comment (andreasKroepelin) [reply themes/bristol.typ:8]: My bad, it is probaby best to leave this in.

**Comment by andreasKroepelin** [comment]
Nice work! Thanks again for your collaboration and also for your patience with my reviews.

**Pull request merged**